app/stores/CollectionsStore.js

```diff
@@ -33,10 +34,24 @@ export default class CollectionsStore extends BaseStore<Collection> {
 
   @computed
   get orderedData(): Collection[] {
-    return filter(
-      naturalSort(Array.from(this.data.values()), "name"),
-      (d) => !d.deletedAt
+    let collections = Array.from(this.data.values());
+
+    collections = collections.filter((collection) =>
+      collection.deletedAt ? false : true
     );
+
+    return collections.sort((a, b) => {
+      if (a.index === b.index) {
+        return 0;
+      }
+      return a.index < b.index ? -1 : 1;
+    });
+  }
+
+  @computed
+  get indexForNew(): string {
+    const collections = this.orderedData;
+    return fractionalIndex(collections[collections.length - 1].index, null);
```

Comment:
See comment above about moving this to the server, we'd have to query the last collection in the team